web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -75,8 +75,13 @@ export class DataStoreService {
   }
 
   // TODO: Define return types for methods in this class
-  updateProjectLayer(projectId: string, layer: Layer) {
-    const { id: layerId, ...layerDoc } = layer;
+  updateProject(projectId: string, layer: Layer) {
+    const { id: layerId, forms, ...layerDoc } = layer;
+    if (forms && !forms.isEmpty()) {
+      layerDoc['forms'] = forms.toJS();
+    } else {
+      delete layerDoc['forms'];
```

Comment:
`forms` will already be extracted from `layerDoc` at like 79, so this else block isn't necessary.